src/client/proxy/matcher.rs

```diff
@@ -358,13 +358,16 @@ fn parse_env_uri(val: &str) -> Option<Intercept> {
     let authority = uri.authority()?;
 
     if let Some((userinfo, host_port)) = authority.as_str().split_once('@') {
-        let (user, pass) = userinfo.split_once(':')?;
-        let user = percent_decode_str(user).decode_utf8_lossy();
-        let pass = percent_decode_str(pass).decode_utf8_lossy();
+        let (user, pass) = match userinfo.split_once(':') {
+            Some((user, pass)) => (user, Some(pass)),
+            None => (userinfo, None),
+        };
+        let user = percent_decode_str(user).decode_utf8_lossy().into_owned();
```

Comment:
Was the `into_owned()` call required? Previously, it seemed we didn't need to create a new intermediary `String` just to encode it back into a header, I think?